app/providers/base.py

```
@@ -0,0 +1,390 @@
+            self._cached_alts = {}
+            self._cached_at = now
+            return {}
+
+        if not self._should_refresh(now):
+            return self._cached_index or {}
+
+        index: Dict[str, str] = {}
+        alts: Dict[str, List[str]] = {}
+        try:
+            index, alts = self._fetch_index_from_url()
+        except Exception as exc:
```

Ruff (0.14.13)

Do not catch blind exception: `Exception`

(BLE001)